rust-runtime/aws-smithy-wasm/src/wasi.rs

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! HTTP WASI Adapter
+use aws_smithy_http::header::ParseError;
+use aws_smithy_runtime_api::{
+    client::{
+        http::{
+            HttpClient, HttpConnector, HttpConnectorFuture, HttpConnectorSettings,
+            SharedHttpClient, SharedHttpConnector,
+        },
+        orchestrator::HttpRequest,
+        result::ConnectorError,
+        runtime_components::RuntimeComponents,
+    },
+    http::Response,
+    shared::IntoShared,
+};
+use aws_smithy_types::body::SdkBody;
+use bytes::{Bytes, BytesMut};
+use wasi::http::{
+    outgoing_handler,
+    types::{self as wasi_http, OutgoingBody, RequestOptions},
+};
+
+/// Creates a connector function that can be used during instantiation of the client SDK
+/// in order to route the HTTP requests through the WebAssembly host. The host must
+/// support the WASI HTTP proposal as defined in the Preview 2 specification.
+pub fn wasi_http_client() -> SharedHttpClient {
```

Comment:
Will this client ever need any kind of configuration?

Reply:
As discussed offline I implemented an empty `WasiHttpClientBuilder` for this so it can be extended with more configuration in a backwards compatible way in the future.